File: config/phoenix/flashspace.js
async function runCommand(command, args = []) {
  return new Promise((resolve, reject) => {
    Task.run(command, args, (task) => {
      if (task.status !== 0) {
        reject(
          new Error(`Command "${command}" failed with status ${task.status}`),
        );
      }

      resolve(task.output.trim());
    });
  });
}

async function getFlashSpaceProfile() {
  return runCommand("/usr/local/bin/flashspace", ["get-profile"]);
}

async function getFlashSpaceWorkspace() {
  return runCommand("/usr/local/bin/flashspace", ["get-workspace"]);
}

File: local/share/window-manager/flashspace.rb
module Flashspace
  class << self
    def profile
      `flashspace get-profile`.strip
    end

    def workspace
      `flashspace get-workspace`.strip
    end

    def apps(workspace, profile)
      `flashspace list-apps #{workspace} --profile #{profile}`.strip.split("\n")
    end
  end
end
